core/src/main/java/io/questdb/cairo/mig/Mig702.java

```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.mig;
+
+import io.questdb.cairo.vm.api.MemoryMARW;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.str.Path;
+
+import static io.questdb.cairo.mig.MigrationUtils.openFileSafe;
+
+class Mig702 {
+    public static final long TX_BASE_OFFSET_A_32 = 8;
+    public static final long TX_BASE_OFFSET_B_32 = 32;
+    public static final long TX_OFFSET_SEQ_TXN_64_MIG = 80;
+    public static final long TX_OFFSET_CHECKSUM_32_MIG = TX_OFFSET_SEQ_TXN_64_MIG + 8;
+    public static final long TX_OFFSET_LAG_TXN_COUNT_32_MIG = TX_OFFSET_CHECKSUM_32_MIG + 4;
+    public static final long TX_OFFSET_LAG_ROW_COUNT_32_MIG = TX_OFFSET_LAG_TXN_COUNT_32_MIG + 4;
+    public static final long TX_OFFSET_LAG_MIN_TIMESTAMP_64_MIG = TX_OFFSET_LAG_ROW_COUNT_32_MIG + 4;
+    public static final long TX_OFFSET_LAG_MAX_TIMESTAMP_64_MIG = TX_OFFSET_LAG_MIN_TIMESTAMP_64_MIG + 8;
+    private static final String TXN_FILE_NAME_MIG = "_txn";
+    private static final long TXN_VERSION_OFFSET_MIG = 0;
+    private static final long TX_OFFSET_MAP_WRITER_COUNT_MIG = 128;
+
+    private static int getInt(MemoryMARW txMemory, long baseOffset, long txOffsetTransientRowCount64) {
+        return txMemory.getInt(baseOffset + txOffsetTransientRowCount64);
+    }
+
+    private static long getLong(MemoryMARW txMemory, long baseOffset, long txOffsetTransientRowCount64) {
+        return txMemory.getLong(baseOffset + txOffsetTransientRowCount64);
+    }
+
+    static int calculateTxnLagChecksum(long txn1, long seqTxn1, int lagRowCount1, long lagMinTimestamp1, long lagMaxTimestamp1, int lagTxnCount1) {
```

Comment:
Yes, that's right, its purpose is to flag garbage in _txn file because perhaps a previous version of DB wrote to the file.

It's not strong, I agree, but it's cheap to calculate.
Do you still think it's worth changing it?

Reply:
Thanks!